packages/web/components/trade-clipboard/trade-route.tsx

```diff
@@ -0,0 +1,111 @@
+import { AppCurrency } from "@keplr-wallet/types";
+import { Dec } from "@keplr-wallet/unit";
+import { Pool } from "@osmosis-labs/pools";
+import { observer } from "mobx-react-lite";
+import { FunctionComponent } from "react";
+import { useStore } from "../../stores";
+import { ChainStore } from "../../stores/chain";
+import { DenomImage } from "../denom";
+import { Tooltip } from "../tooltip";
+import { TradeRouteProps } from "./types";
+
+function getDenomsFromPool(chainStore: ChainStore, pool: Pool) {
```

Comment:
Interesting way to organise code in file-private functions.

Reply:
I like creating pure functions to make it easier to break down the code and debug it.